src/EnergyPlus/PollutionModule.cc

```diff
@@ -6331,182 +6340,180 @@ void GetFuelFactorInfo(EnergyPlusData &state,
     fuelFactorScheduleUsed = false;
     ffScheduleIndex = 0;
 
-    {
-        auto const &SELECT_CASE_var(fuelName);
-
-        if ((SELECT_CASE_var == "NaturalGas") || (SELECT_CASE_var == "Gas")) {
-            if (Pollution.NatGasCoef.FuelFactorUsed) {
-                fuelFactorUsed = true;
-                fuelSourceFactor = Pollution.NatGasCoef.Source;
-                if (Pollution.NatGasCoef.SourceSched == 0) {
-                    fuelFactorScheduleUsed = false;
-                } else {
-                    fuelFactorScheduleUsed = true;
-                    ffScheduleIndex = Pollution.NatGasCoef.SourceSched;
-                }
+    switch (fuelName) {
+    case Constant::eFuel::NaturalGas: {
+        if (Pollution.NatGasCoef.FuelFactorUsed) {
```

Comment:
I feel like we should be able to work these little blocks into a reusable worker function.  Nothing for now, but could be later.

Reply:
📌